lib/PhpStatistics.php

```diff
@@ -64,4 +66,62 @@ protected function cleanVersion($version) {
 		}
 		return $version;
 	}
+
+	/**
+	 * Get status information about the cache from the OPcache extension
+	 *
+	 * @return array with an array of state information about the cache instance
+	 */
+	protected function getOPcacheStatus(): array {
+		// Test if the OPcache module is installed
+		if (!extension_loaded('Zend OPcache')) {
+			// module not loaded, returning back empty array to prevent any errors on JS side.
+			return [];
+		}
+
+		// get status information about the cache
+		$status = opcache_get_status(false);
+
+		if ($status !== false) {
+			return $status;
+		} else {
+			// no array, returning back empty array to prevent any errors on JS side.
+			return [];
+		}
+	}
+
+	/**
+	 * Get status information about the cache from the APCu extension
+	 *
+	 * @return array with an array of state information about the cache instance
+	 */
+	protected function getAPCuStatus(): array {
```

Comment:
```
	/**
	 * Get status information about the cache from the APCu extension
	 *
	 * @return array with an array of state information about the cache instance
	 */
	protected function getAPCuStatus(): array {
		// Test if the APCu module is installed
		if (!extension_loaded('apcu')) {
			// module not loaded, returning back empty array to prevent any errors on JS side.
			return [
				'cache' => false,
				'sma' => false
			];
		}

		return [
			'cache' => apcu_cache_info(true),
			'sma' => apcu_sma_info(true)
		];
	}
```

Looks a bit easier to me. What do you think?

Reply:
Yes it does. But doesn't this result in mixing return types for parts of the array again?

Reply:
> Yes it does. But doesn't this result in mixing return types for parts of the array again?

That's okay as long as `getAPCuStatus` returns an array.

Reply:
I would prefer to return 'N/A' instead of false in those cases then, as I proposed in the first place. What do you think?

Reply:
Fine by me.